lucene/src/test/java/org/hibernate/hql/lucene/test/UntypedLuceneQueryParsingTest.java

```diff
@@ -124,9 +129,13 @@ private TestFieldBridgeProvider() {
 			Map<String, FieldBridge> indexedEntityBridges = new HashMap<String, FieldBridge>();
 
 			indexedEntityBridges.put( "id", new TwoWayString2FieldBridgeAdaptor( new StringBridge() ) );
-			indexedEntityBridges.put( "name", new NullEncodingTwoWayFieldBridge( new TwoWayString2FieldBridgeAdaptor( new StringBridge() ), "_null_" ) );
+
+			TwoWayFieldBridge fieldBridge = new TwoWayString2FieldBridgeAdaptor( new StringBridge() );
+			NullMarkerCodec stringNullCodec = new LuceneStringNullMarkerCodec( new ToStringNullMarker( "_null_" ) );
+			indexedEntityBridges.put( "name", new NullEncodingTwoWayFieldBridge( fieldBridge, stringNullCodec ) );
 			indexedEntityBridges.put( "position", NumericFieldBridge.LONG_FIELD_BRIDGE );
-			indexedEntityBridges.put( "code", new NullEncodingTwoWayFieldBridge( NumericFieldBridge.LONG_FIELD_BRIDGE, "_null_" ) );
+
+			indexedEntityBridges.put( "code", new NullEncodingTwoWayFieldBridge( NumericFieldBridge.LONG_FIELD_BRIDGE, stringNullCodec ) );
```

Comment:
I'm not sure about what was going on previously here, but I know that we currently do not support having a non-numeric null marker on an otherwise numeric field. At least the default numeric field bridges will only create numeric null markers. So this kind of setup (`"_null_"` used as a marker on a numeric field) cannot happen in practice when you're using Hibernate Search.

To be in line with real-world use cases, you should use a `LuceneLongNullMarkerCodec` here, since `code` seems to be a numeric (long) field. Also, the null marker should be retrieved from the bridge itself. Something like `NumericFieldBridge.LONG_FIELD_BRIDGE.createNullMarker( "-1" )`.

The related code in Hibernate Search is in `org.hibernate.search.engine.nulls.impl.LuceneMissingValueStrategy.createNullMarkerCodec(Class<?>, DocumentFieldPath, NullMarker)` and `org.hibernate.search.engine.metadata.impl.AnnotationMetadataProvider.createNullMarker(FieldBridge, String, DocumentFieldPath)` (both called by `org.hibernate.search.engine.metadata.impl.AnnotationMetadataProvider.determineNullMarkerCodec(FieldBridge, Field, ConfigContext, ParseContext, Class<?>, DocumentFieldPath)`).